### ISSUE TITLE
handle unavailability

### ISSUE THREAD

**Issue by bboehmke** [mvisonneau/gitlab-ci-pipelines-exporter#29]
If the Gitlab get unavailable (e.g. caused by an update) while exporter is running the exporter will exit:

```
gitlab-ci-pipelines-exporter_1  | time="2019-10-19T04:07:09Z" level=error msg="ListProjectPipelines: Get https://gitlab/api/v4/projects/276/pipelines?ref=master: dial tcp 10.169.66.61:443: connect: no route to host"
gitlab-ci-pipelines-exporter_1  | panic: runtime error: index out of range [0] with length 0
gitlab-ci-pipelines-exporter_1  |
gitlab-ci-pipelines-exporter_1  | goroutine 556 [running]:
gitlab-ci-pipelines-exporter_1  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/cmd.(*Client).pollProjectRef(0xc00014a2e8, 0xc0003d58c0, 0xc0006d8910, 0x6)
gitlab-ci-pipelines-exporter_1  |       /build/cmd/gitlab.go:275 +0xf30
gitlab-ci-pipelines-exporter_1  | created by github.com/mvisonneau/gitlab-ci-pipelines-exporter/cmd.(*Client).pollProject
gitlab-ci-pipelines-exporter_1  |       /build/cmd/gitlab.go:244 +0x595
```

Is it possible to handle such unavailability inside the application?
Otherwise the current workaround is docker-compose.yml with `restart: always` which results in a lot of log output.

**Comment by mvisonneau** [comment]
yes, that would be nice to handle this situation more properly! Thanks for the raising the issue @bboehmke.

**Comment by agilob** [comment]
@mvisonneau Won't this be fixed by my MR from a few weeks ago?

```
	for {
		time.Sleep(b.Duration())

		pipelines, _, err := c.Pipelines.ListProjectPipelines(gp.ID, &gitlab.ListProjectPipelinesOptions{Ref: gitlab.String(ref)})
		if err != nil {
			log.Errorf("ListProjectPipelines: %s", err.Error())
			continue
		}

		if len(pipelines) == 0 {
			log.Debugf("Could not find any pipeline for %s:%s", gp.PathWithNamespace, ref)
		} else if lastPipeline == nil || lastPipeline.ID != pipelines[0].ID || lastPipeline.Status != pipelines[0].Status {
```

Seems like @bboehmke is using older version without check  for `len(pipelines) == 0`

Just make a new release ;)

**Comment by mvisonneau** [comment]
indeed 🤦‍♂ ! I just released `0.2.9` that should sort this out. Feel free to reopen if it is still not the case.
